apps/omg_watcher/lib/web/controllers/transaction.ex

```diff
@@ -92,19 +90,47 @@ defmodule OMG.Watcher.Web.Controller.Transaction do
   defp respond({:error, code}, conn) when is_atom(code), do: handle_error(conn, code)
 
   defp parse_request_body(%{"inputs" => inputs, "outputs" => outputs}) when is_list(inputs) and is_list(outputs) do
-    {
+    number_of_currencies =
       inputs
-      |> Enum.map(&Map.delete(&1, "txbytes"))
-      |> Enum.map(fn %{} = input ->
-        input = input |> Enum.into(%{}, fn {k, v} -> {String.to_existing_atom(k), v} end)
-        %{input | currency: Base.decode16!(input.currency, case: :mixed)}
-      end),
-      outputs
-      |> Enum.map(fn %{} = output ->
-        output = output |> Enum.into(%{}, fn {k, v} -> {String.to_existing_atom(k), v} end)
-        %{output | owner: OMG.API.Crypto.decode_address!(output.owner)}
-      end)
-    }
+      |> Enum.map(fn %{"currency" => currency} -> currency end)
+      |> Enum.dedup()
+      |> Enum.count()
+
+    cond do
```

Comment:
these checks and processing look a bit out of place here, why?

Reply:
Since deeper in the code we allow for multiple currencies I think that's the right place to put those validation. If we would not validate that here, we would break the API of this endpoint.

Reply:
`State.Transaction.create_from_utxos` seems like the right place, and that comes before `parse...`, so why not there?

Reply:
wait, for example: `:currency_mixing_not_possible` is actually thrown there so this is a duplicate check, so is `at_least_one_input_required`. Am I missing something?

Reply:
You are right. My mistake.

Reply:
It turned out I actually need to validate number of inputs right there in the controller. I need to get a currency. In endpoint API currency is tied to inputs. I need to extract it from inputs and put into outputs.